src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -856,14 +857,17 @@ public void recalculateIsland(@NotNull Island island) {
     private void recalculateIsland(@NotNull Island island, @NotNull List<Chunk> chunks) {
         chunks.stream().map(chunk -> chunk.getChunkSnapshot(true, false, false)).forEach(chunk -> {
             World world = Bukkit.getWorld(chunk.getWorldName());
+            boolean oceanWorld = IridiumSkyblockAPI.getInstance().getGeneratorType() == GeneratorType.OCEAN;
             int maxHeight = world == null ? 255 : world.getMaxHeight() - 1;
+
             for (int x = 0; x < 16; x++) {
                 for (int z = 0; z < 16; z++) {
                     if (island.isInIsland(x + (chunk.getX() * 16), z + (chunk.getZ() * 16))) {
                         final int maxy = Math.min(maxHeight, chunk.getHighestBlockYAt(x, z));
                         for (int y = LocationUtils.getMinHeight(world); y <= maxy; y++) {
                             XMaterial material = XMaterial.matchXMaterial(chunk.getBlockType(x, y, z));
-                            if (material.equals(XMaterial.AIR)) continue;
+                            if (material == XMaterial.AIR) continue;
+                            if (oceanWorld && material == IridiumSkyblock.getInstance().getChunkGenerator().getMainMaterial(world)) return;
```

Comment:
Why are we specifying if its oceanWorld here, and why are we returning?

Reply:
We use the oceanWorld boolean here to speed up the recalculation. Checking that often if the material is in the set (or list?) will probably be pretty bad. But yeah, the return shouldn't be there

Reply:
And we should ignore air in every world